### PR TITLE
Lmedavid/feature/http basic

### PR DESCRIPTION
**Pull request by laurentdavid**
Hi,

I have added (or resurfaced) the http basic authentication parameter for the commands such as run_site_diff.sh or  run_url_pair_diff.sh.

It uses two additional parameter to get through http basic authentication : http_username and http_password.
I have tried to do as little change as possible.

There is still an open question ref.  run_url_pair_diff.sh, how to manage where credentials are different from one site to another.

Cheers,

Laurent

### CODE REVIEW

File: dpxdt/client/capture.js [hunk 1]
```diff
@@ -77,6 +76,12 @@ if (config.cookies) {
     });
 }
 
+// Add username and password as a parameter for HTTP basic auth
+if (config.httpUserName && config.httpPassword) {
```
Comment (bslatkin) [inline dpxdt/client/capture.js]: Would you mind adding documentation of these parameters to README.md?

Comment (laurentdavid) [inline dpxdt/client/capture.js]: Thanks for your reply. Indeed, I will  do it.

File: dpxdt/tools/site_diff.py [hunk 2]
```diff
@@ -228,6 +228,12 @@ def run(self,
         good_urls = set()
 
         yield heartbeat('Scanning for content')
+        
+        http_username=None
```
Comment (bslatkin) [inline dpxdt/tools/site_diff.py]: Style nit, can you add spaces around the assignment?

http_username = None
http_password = None

File: dpxdt/tools/site_diff.py [hunk 3]
```diff
@@ -228,6 +228,12 @@ def run(self,
...
+        http_password=None
+        if FLAGS.http_username and FLAGS.http_password:
```
Comment (bslatkin) [inline dpxdt/tools/site_diff.py]: Alternatively, you can remove this if statement entirely. We may want to support username with blank passwords? So perhaps the right approach is just:

```
http_username = FLAGS.http_username
http_password = FLAGS.http_password
```

And let the values default to `None`

File: dpxdt/tools/site_diff.py [hunk 4]
```diff
@@ -238,7 +244,7 @@ def run(self,
             seen_urls.update(pending_urls)
             yield heartbeat(
                 'Scanning %d pages for good urls' % len(pending_urls))
-            output = yield [fetch_worker.FetchItem(u) for u in pending_urls]
+            output = yield [fetch_worker.FetchItem(u,username=http_username,password=http_password) for u in pending_urls]
```
Comment (bslatkin) [inline dpxdt/tools/site_diff.py]: Can you rewrap this following [PEP8](https://www.python.org/dev/peps/pep-0008/)

```
            output = yield [fetch_worker.FetchItem(
                                url,
                                username=http_username,
                                password=http_password)
                            for url in pending_urls]
```

File: dpxdt/tools/site_diff.py [hunk 5]
```diff
@@ -293,6 +299,11 @@ def run(self,
             if FLAGS.cookies:
                 config_dict['cookies'] = json.loads(
                     open(FLAGS.cookies).read())
+            if http_username and http_password:
```
Comment (bslatkin) [inline dpxdt/tools/site_diff.py]: May be worth splitting this into two if statements so we can support http_username being set but http_password not being set?

File: dpxdt/tools/url_pair_diff.py [hunk 6]
```diff
@@ -88,6 +88,11 @@ def run(self,
             config_dict['injectCss'] = FLAGS.inject_css
         if FLAGS.inject_js:
             config_dict['injectJs'] = FLAGS.inject_js
+
+	if FLAGS.http_username and FLAGS.http_password:
```
Comment (bslatkin) [inline dpxdt/tools/url_pair_diff.py]: Same thing here about having two if statements?

File: dpxdt/client/capture.js (hunk 1 above)
Comment (bslatkin) [inline dpxdt/client/capture.js]: Same comment (I have below) about allowing httpUserName to be specified but httpPassword to be empty. May be worth splitting this into two if statements and assignments.

**Comment by bslatkin** [comment]
Thanks for doing this! I left some inline comments. Hope you don't mind and thanks in advance.

**Comment by laurentdavid** [comment]
Hi,

I have made the changes required on my branch. 

Laurent

**Comment by laurentdavid** [comment]
Hi Brett,

It seems that my version is getting quickly outdated by recent commits. 

Any chance to merge it soon ? 

Laurent

File: dpxdt/client/capture.js (hunk 1 above)
Comment (bslatkin) [inline dpxdt/client/capture.js]: Oh cool I meant for you to add it here to the documentation of `config.json` <https://github.com/bslatkin/dpxdt#format-of-config>

**Pull request merged**